### ISSUE TITLE
Tar cannot change ownership to uid 0, gid 0

### ISSUE THREAD

**Issue by kuon** [vn971/rua#89]
I am not sure if this is rua's fault (because of the permission architecture it uses), or if the PKGBUILD of the package must be changed.

The package I am trying to install is:

https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=ms-teams

And I get the following errors (truncated, as there is the same error for all files in the archive):

```
tar: ./usr/share/teams-insiders/version: Cannot change ownership to uid 0, gid 0: Invalid argument
tar: ./usr/share/teams-insiders: Cannot change ownership to uid 0, gid 0: Invalid argument
tar: ./usr/share: Cannot change ownership to uid 0, gid 0: Invalid argument
tar: ./usr: Cannot change ownership to uid 0, gid 0: Invalid argument
tar: .: Cannot change ownership to uid 0, gid 0: Invalid argument
tar: Exiting with failure status due to previous errors
==> ERROR: A failure occurred in package().
    Aborting...
Build failed with exit code 4 in /home/kuon/.cache/rua/build/ms-teams
```

Adding `--no-same-owner` to the tax extract command fix it, but I don't know if rua ought to support the PKGBUILD as it is.

**Comment by vn971** [comment]
I think the PKGBUILD is broken, because packages are by design built from non-root user. It should thus not be capable of creating root-owned files -- at all, ever.
I think it won't even work with a vanilla `makepkg -f`, though I haven't tried.

Closing for now, feel free to re-open if you're sure it builds with `makepkg -f` and we need to make some changes to build it as well.

**Comment by Morganamilo** [comment]
Makepkg runs package in fakeroot so that root owned files can be created.

**Comment by kuon** [comment]
Yes, it does work with `makepkg` but I wasn't sure if this particular scenario should be supported by rua or if we should patch the `PKGBUILD`.

**Comment by vn971** [comment]
Hmmm, that's interesting. I've just tried to build `ms-teams`, and it builds fine for me. I used `rua install ms-teams` to do it.

```
In order to install all targets, the following pacman packages will need to be installed:
  gnome-keyring
  libgnome-keyring

And the following AUR packages will need to be built and installed:
  ms-teams

Proceed? [O]=ok, Ctrl-C=abort. 
Proceed? [O]=ok, Ctrl-C=abort. o

Reviewing "/home/vasya/.config/rua/pkg/ms-teams". 
[D]=view upstream changes since your last review, [M]=accept/merge upstream changes, [S]=(shellcheck not available until you merge), [T]=run shell to edit/inspect, [O]=(cannot use the package until you merge) d

...

shellcheck found no problems

Reviewing "/home/vasya/.config/rua/pkg/ms-teams". 
[S]=run shellcheck on PKGBUILD, [D]=(identical to upstream, empty diff), [T]=run shell to edit/inspect, [O]=ok, use package o
Packages need to be installed:

    pacman -S --asdeps --needed 'gnome-keyring' 'libgnome-keyring'

Enter S to `sudo` install it, or install manually and press M when done. m
==> Making package: ms-teams 1.2.00.30358-1 (2019-11-12T10:16:01 CET)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading teams-insiders_1.2.00.30358_amd64.deb...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 62.2M  100 62.2M    0     0  1538k      0  0:00:41  0:00:41 --:--:-- 1797k
==> Validating source files with sha256sums...
    teams-insiders_1.2.00.30358_amd64.deb ... Passed
==> Extracting sources...
  -> Extracting teams-insiders_1.2.00.30358_amd64.deb with bsdtar
==> Entering fakeroot environment...
==> Starting package()...
  -> Extracting the data.tar.xz
==> Tidying install...
  -> Removing empty directories...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Removing static library files...
  -> Compressing man and info pages...
==> Checking for packaging issues...
==> Creating package "ms-teams"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Adding install file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Leaving fakeroot environment.
==> Finished making: ms-teams 1.2.00.30358-1 (2019-11-12T10:17:20 CET)
Package ms-teams-1.2.00.30358-1-x86_64.pkg.tar.xz has no SUID files.
[E]=list executable files, [L]=list all files, [T]=run shell to inspect, [I]=show install file, [O]=ok, proceed. 
```

Need to think about it a bit more.

**Comment by vn971** [comment]
@kuon  Do you have any "wrap_args" configured in RUA on your system?

**Comment by kuon** [comment]
No I have no `wrap_args` configured.

**Comment by ionphractal** [comment]
I have the same issue, also no wrap args. No luck with `rua install ms-teams` nor any other packages I've tried so far that use tar in `package()` function (e.g. keybase-bin, signal-desktop-bin), which is executed under fakeroot.

Not sure if that is replicable for you, but running the rua container with fakeroot produces the same error for me. For example if you downloaded the rua-0.15.1.tar.gz release, execute in the same folder:
```
bash ~/.config/rua/.system/wrap.sh --bind $(pwd) $(pwd) fakeroot tar xzf rua-0.15.1.tar.gz
tar: rua-0.15.1/.editorconfig: Cannot change ownership to uid 0, gid 0: Invalid argument
[...]
tar: rua-0.15.1: Cannot change ownership to uid 0, gid 0: Invalid argument
tar: Exiting with failure status due to previous errors
```
So it looks to me like some sort of incompatibility between bubblewrap and fakeroot. I found some similar issues while searching for the cause:

https://github.com/bazelbuild/bazel/issues/5096
https://github.com/containers/bubblewrap/issues/210

Funny thing: if you use `--uid 0 --gid 0` with the command above like in the last issue link then fakeroot works. I don't think this poses a real security threat because it is (or at least should be...) just the user mapping and no real root elevation but fakeroot seems to be happy about it. Of course this only applies if there is no bug in the bubblewrap or kernel code that handles namespaces... but at least the possibility should be low.

Thing is just that makepkg will think it is running as root, so seems it can't be used as a workaround in wrap.sh or as wrap_args...

**Comment by vn971** [comment]
Very interesting... BTW, do you have `bubblewrap` package installed, or `bubblewrap-suid`? (I have the latter because I'm running on `linux-hardened` kernel, which doesn't have `userns`.)

**Comment by kuon** [comment]
I have `bubblewrap 0.3.3-3`

**Comment by ionphractal** [comment]
I had also `extra/bubblewrap 0.3.3-3` installed, switched to `bubblewrap-suid` and no issues - even though I use the normal linux kernal in Arch. 

And it is no wonder suid works because it is actually using root permissions (https://en.wikipedia.org/wiki/Setuid).

```
~ % ls -la /usr/bin/bwrap
-rwsr-xr-x 1 root root 55160 31. Jul 11:44 /usr/bin/bwrap
```

I feel like it gives the security concept a little crack. But I guess even though the container wrapper is executed as root, the container doesn't(/shouldn't if there is no bug) have root inside and additionally basically everything is mounted read-only by rua. But fact is there is something different in permissions inside the bwrap container.

**Comment by ionphractal** [comment]
Seems like this bug https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=909727 and yes: the presented workaround works:
```
FAKEROOTDONTTRYCHOWN=true bash ~/.config/rua/.system/wrap.sh --bind $(pwd) $(pwd) fakeroot tar xzf rua-0.15.1.tar.gz
echo $?
0
```

**Comment by vn971** [comment]
@ionphractal very interesting indeed. Thanks for digging it up and explaining.

Bubblewrap being root (suid) is a known thing, yeah. It conceptually works by preparing a namespace jail, dropping off rights and then running the requested program with user permissions. It is believed by the tool that this is safer than enabling `userns`, which had great many CVE-s in the past. ArchLinux, however, switched to enable userns on mainstream `linux` and `linux-lts` a few months ago, so we (or I) should kinda adapt to that.

**Comment by vn971** [comment]
I wonder what I need to do with the issue now. Options that I see:

1. mark it as a bug and leave it hanging in the issue tracker documenting the thing

2. pass `FAKEROOTDONTTRYCHOWN=true`. It generally should be OK for most builds that I can imagine. But just because of how terrible hack that is, I'm seriously considerate on doing that..

**Comment by ionphractal** [comment]
For me personally I just added the statement to a `.config/rua/wrap_args.d/fakeroot.sh` file which is a normal bash script that is sourced before bwrap is called -> not nice but works. In any case I would be document it in the README because it is quite an obvious yet unintuitive issue.

But despite that: providing two packages (or something like a double-package like bubblewrap, see itself https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/bubblewrap)
* one for bubblewrap-suid and 
* one without that includes such a wrap_args.d file by default or any appropriate fix, 
might be worth considering. How does that sound to you?

**Comment by vn971** [comment]
@ionphractal regarding implementation of the thing. I think you can put such a logic even right in wrap.sh script:
```
if test -u "$(which bwrap)"; then
  export ...=...
fi
```
This has the advantage that it'll work even if the user changes kernels during reboots.

The main question for me, however, is to whether have this env variable at all or not.. I think I'll limit the actions to documenting in README for now, and see if many more reports like that will pop out.

**Comment by ionphractal** [comment]
Yes it could be in the wrap.sh script but you would have change your template and create a new version, which is why I put it into the sourced file. It would at least need to be before the source so that in worst case it could be overwritten.

Can't tell you what action suits you best but having it documented would be a great start I think.

**Comment by kuon** [comment]
Just for reference of other packages that fails to install:

- `google-earth` (`bsdtar` `./usr/bin/google-earth: Can't set user=0/group=0 for usr/bin/google-earth`)
- `chatty` (`bsdcpio: Can't set user=0/group=0 for LICENSE`)

**Comment by kuon** [comment]
As the debian bug was open since 2018, do you have an idea when/how we could get this fixed? I am not familiar with debian development process. The fix seems trivial, handling `EINVAL`.

This seems an important enough issue to have it fixed asap.

One alternative would be to fix fakeroot and ship a patched version with rua, this would be miles better than `FAKEROOTDONTTRYCHOWN=true` which could break packages that require some user/group to be set.

**Comment by vn971** [comment]
> would be miles better than `FAKEROOTDONTTRYCHOWN=true` which could break packages that require some user/group to be set.

On the other hand, you should never rely on owners inside a *.pkg.tar package anyway, right? I mean, what would be a scenario where you need a specific owner/group inside a file in `pkgdest`?

I'll ask in #archlinux-aur as well, just in case.

**Comment by vn971** [comment]
UPD: this statement is not correct. Ownership, when set in `prepare()` or `package()`, is preserved during installation.

**Comment by vn971** [comment]
The proper work-around seems to be doing the `untar` / `bsdtar -x` command inside `prepare()` function of PKGBUILD, not in `package()`. This is proper from makepkg-s design and it also helps us avoiding the bug.

I do not have a patch to fix fakeroot as well, so that approach is not possible for me. If there'll be such a patch, we can submit it to fakeroot for upstream merging as well I think.

I'll ask some individual packages to move the unpacking step into `prepare()`. This should fix it for those specific packages. Meanwhile open to other ideas or PR-s.

**Comment by vn971** [comment]
I've documented X11 access restriction and fakeroot bug situation [here](https://github.com/vn971/rua#limitations) and [here](https://github.com/vn971/rua/blob/master/docs/x11access.md)

**Comment by qoh** [comment]
With the package `v4l2loopback-dkms` I get a different but somewhat similar error:
```
==> Entering fakeroot environment...
==> Starting package()...
'COPYING' -> '/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/share/licenses/v4l2loopback/COPYING'
install -p -m 755 -d "/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/bin"
install -p -m 755 utils/v4l2loopback-ctl "/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/bin"
help2man -N --name "control v4l2 loopback devices" utils/v4l2loopback-ctl > man/v4l2loopback-ctl.1
install -p -m 755 -d "/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/share/man/man1"
install -m 644 man/v4l2loopback-ctl.1 "/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/share/man/man1"
cp: failed to preserve ownership for '/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/src/v4l2loopback-0.12.5/doc': Invalid argument
cp: failed to preserve ownership for '/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/src/v4l2loopback-0.12.5/examples': Invalid argument
cp: failed to preserve ownership for '/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/src/v4l2loopback-0.12.5/man': Invalid argument
cp: failed to preserve ownership for '/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/src/v4l2loopback-0.12.5/tests': Invalid argument
cp: failed to preserve ownership for '/home/ns/.cache/rua/build/v4l2loopback-dkms/pkg/v4l2loopback-dkms/usr/src/v4l2loopback-0.12.5/utils': Invalid argument
```
It indeed also builds with `FAKEROOTDONTTRYCHOWN=true`.

**Comment by mqus** [comment]
Same for me with zfs-dkms. rua gives the output below while makepkg works without any issues. 
```
$ rua install zfs-dkms
WARNING: your AUR repo is up-to-date.
If you continue, the build directory will be removed and the build will be re-run.
If you don't want that, consider resolving the situation manually,
for example:    rua builddir '/home/mqus/.cache/rua/build/zfs-dkms'


Reviewing "/home/mqus/.config/rua/pkg/zfs-dkms". 
[S]=run shellcheck on PKGBUILD, [D]=(identical to upstream, empty diff), [T]=run shell to edit/inspect, [O]=ok, use package. O
==> Making package: zfs-dkms 0.8.4-2 (Thu 24 Sep 2020 23:39:29 CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading zfs-0.8.4.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   141  100   141    0     0    691      0 --:--:-- --:--:-- --:--:--   694
100   636  100   636    0     0   1586      0 --:--:-- --:--:-- --:--:--  1586
100 9068k  100 9068k    0     0  3063k      0  0:00:02  0:00:02 --:--:-- 4453k
  -> Downloading zfs-0.8.4.tar.gz.asc...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   145  100   145    0     0    823      0 --:--:-- --:--:-- --:--:--   823
100   640  100   640    0     0   1748      0 --:--:-- --:--:-- --:--:--  1748
100   836  100   836    0     0   1006      0 --:--:-- --:--:-- --:--:--  1006
  -> Downloading d90015c5439167456d02585c25147742453d9aa6.patch...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5936  100  5936    0     0  22656      0 --:--:-- --:--:-- --:--:-- 22570
  -> Found 0001-only-build-the-module-in-dkms.conf.patch
==> Validating source files with sha256sums...
    zfs-0.8.4.tar.gz ... Passed
    zfs-0.8.4.tar.gz.asc ... Skipped
    d90015c5439167456d02585c25147742453d9aa6.patch ... Passed
    0001-only-build-the-module-in-dkms.conf.patch ... Passed
==> Validating source files with b2sums...
    zfs-0.8.4.tar.gz ... Passed
    zfs-0.8.4.tar.gz.asc ... Skipped
    d90015c5439167456d02585c25147742453d9aa6.patch ... Passed
    0001-only-build-the-module-in-dkms.conf.patch ... Passed
==> Verifying source file signatures with gpg...
    zfs-0.8.4.tar.gz ... Passed
==> Extracting sources...
  -> Extracting zfs-0.8.4.tar.gz with bsdtar
==> Starting prepare()...
patching file config/kernel-kmem-vmalloc.m4
patching file config/kernel.m4
patching file include/spl/sys/kmem.h
patching file module/spl/spl-kmem-cache.c
patching file module/spl/spl-kmem.c
patching file scripts/dkms.mkconf
Hunk #1 succeeded at 25 with fuzz 2.
Hunk #2 succeeded at 59 with fuzz 1 (offset -4 lines).
autoreconf: configure.ac: AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'config'.
libtoolize: copying file 'config/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'config'.
libtoolize: copying file 'config/libtool.m4'
libtoolize: copying file 'config/ltoptions.m4'
libtoolize: copying file 'config/ltsugar.m4'
libtoolize: copying file 'config/ltversion.m4'
libtoolize: copying file 'config/lt~obsolete.m4'
configure.ac:49: installing 'config/compile'
configure.ac:42: installing 'config/missing'
==> Starting build()...
==> Entering fakeroot environment...
==> Starting package()...
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/sys/crypto': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/sys/lua': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/sys/sysevent': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/sys/fm/fs': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/sys/fm': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/sys/fs': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/sys': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/spl/rpc': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/spl/sys': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/spl': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include/linux': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/include': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/zcommon': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/lua/setjmp': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/lua': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/spi': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/os': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/asm-x86_64/sha2': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/asm-x86_64/aes': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/asm-x86_64/modes': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/asm-x86_64/sha1': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/asm-x86_64': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/api': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/io': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/core': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/algs/sha2': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/algs/edonr': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/algs/aes': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/algs/modes': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/algs/sha1': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/algs/skein': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/algs': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/include/sha2': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/include/aes': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/include/sys/crypto': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/include/sys/ia32': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/include/sys': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/include/modes': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/include/sha1': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp/include': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/icp': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/zfs': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/unicode': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/avl': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/spl': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module/nvpair': Invalid argument
cp: failed to preserve ownership for '/home/mqus/.cache/rua/build/zfs-dkms/pkg/zfs-dkms/usr/src/zfs-0.8.4/module': Invalid argument
==> ERROR: A failure occurred in package().
    Aborting...
Build failed with exit code 4 in /home/mqus/.cache/rua/build/zfs-dkms
```

**Comment by vn971** [comment]
Sorry for the slow reply. It seems there's a number of packages affected, especially considering that not everyone even raises a bug report or writes a comment if something off happens (thanks everyone who does!)

With that in mind, I think it's proper to apply the `FAKEROOTDONTTRYCHOWN=true` fix after all.

Made the change now, released in the latest version (0.17.13) :arrow_up:
